dtale/views.py

```diff
@@ -172,19 +172,16 @@ class DtaleData(object):
     """
 
     def __init__(self, data_id, url):
+        if data_id != None:
+            data_id = int(data_id)
         self._data_id = data_id
         self._url = url
-        self._name_or_data_id = global_state.convert_name_to_url_path(
-            (global_state.get_metadata(self._data_id) or {}).get("name")
-        )
-        if self._name_or_data_id is None:
-            self._name_or_data_id = self._data_id
         self._main_url = self.build_main_url()
         self._notebook_handle = None
         self.started_with_open_browser = False
 
     def build_main_url(self, data_id=None):
-        return "{}/dtale/main/{}".format(self._url, data_id or self._name_or_data_id)
+        return "{}/dtale/main/{}".format(self._url, self._data_id)
```

Comment:
I think this logic needs to be revisited.  I just tried executing the following:
```
import dtale
import pandas as pd

dtale.show(pd.DataFrame(dict(a=[1,2,3], b=[4,5,6])), name='foo')
```
Then I tried hitting the url `http://localhost:40000/dtale/main/foo` and hit this:
![image](https://user-images.githubusercontent.com/11547371/108018964-21cacf00-6fe7-11eb-8385-62fdfd111db4.png)

I doubt people use this feature very much but it is a "nice to have" that would be lost with this

Reply:
God, I can't believe I didn't have a test for this 🤦

Reply:
It is nice to have. However, since the same name can be used on multiple data_ids and find_data_id() will return the first matched data_id, it's probably not that useful at all. I'm going for a dedicated route approach like /dtale/main/name/foo.

Reply:
added in new commit